### ISSUE TITLE
[Bug] Once the region discovery is done calls with WithAzureRegion set to false should not hit the regional token endpoint

### ISSUE THREAD

**Issue by neha-bhargava** [AzureAD/microsoft-authentication-library-for-dotnet#2260]
**Which Version of MSAL are you using ?**

Latest

**Repro**

```csharp
// 1st call with regional auth set to true
await cca.AcquireTokenForClient(scopes) .WithAzureRegion(true);

// 2nd call to global
await cca.AcquireTokenForClient(scopes) .WithAzureRegion(false);
```

**Expected behavior**
The 2nd call should go to global token endpoint.

**Actual behavior**
The 2nd call is going to regional token endpoint.

**Possible Solution**
This is happening because we are caching the network metadata with preferred network as the regional endpoint once the region is discovered for both the original authority and the authority created with the region.

Either cache the regional metadata separate, or cache it only for the regional authority.

**Comment by neha-bhargava** [comment]
Marking it as P1 since the network metadata cache is a static cache and once the region is discovered we will be hitting the regional endpoint irrespective of the value for WithAzureRegion. This could cause problems if someone wants to fallback to global in case ests-r endpoint fails or if someone wants to rollout traffic to ests-r in phases.

**Comment by jmprieur** [comment]
Reopenining the issue closed by the bot (until we release it)

**Comment by pmaytak** [comment]
This is included in [MSAL 4.25.0 release](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.25.0).
